folly/random/xoshiro256pp.h

```diff
@@ -95,35 +86,21 @@ class xoshiro256pp {
   }
 
  private:
-  using vector_type = VectorType;
+  using vector_type = uint64_t;
   static constexpr uint64_t StateSize = 4;
-  static constexpr uint64_t VecResCount = 8;
-  static constexpr uint64_t ResultCount =
-      VecResCount * (sizeof(vector_type) / sizeof(result_type));
-  union {
-    vector_type vecRes[VecResCount]{};
-    result_type res[ResultCount];
-  };
-  vector_type state[VecResCount][StateSize]{};
+  static constexpr uint64_t VecResCount = 16;
```

Comment:
This, plus the elimination of the vector_type choice means it's actually cutting the buffer size in half under x86_64, which we don't want. 

Any idea what the auto-vectorization looks like for this new version on x86_64? IIRC the original was a bit annoying to get right when I first wrote this.